collect_app/src/androidTest/java/org/odk/collect/android/support/matchers/ProjectMatcher.kt

```diff
@@ -0,0 +1,30 @@
+package org.odk.collect.android.support.matchers
+
+import android.view.View
+import android.widget.TextView
+import androidx.test.espresso.matcher.BoundedMatcher
+import androidx.test.espresso.matcher.ViewMatchers.assertThat
+import org.hamcrest.Description
+import org.hamcrest.Matcher
+import org.hamcrest.Matchers.`is`
+import org.odk.collect.android.R
+import org.odk.collect.android.projects.Project
+
+object ProjectMatcher {
```

Comment:
I'm not sure if I like it because such a function would just check if both texts are displayed but it doesn't check where, so theoretically we may end up with mixed icon and project name and it would still pass right?

Reply:
It would check that both pieces of text are within the same parent view (`hasChild` only checks direct children I think) so I think it would protect against the case you're thinking of? My main goal was just to try and avoid view IDs or any knowledge of the view structure in the test.

Reply:
> so I think it would protect against the case you're thinking of?

I don't think so. It would be called on a layout (project_list_item) which contains icon and project name (both TextViews) but if you don't use ids you can't be sure which element contains which value. You just confirm that both values exists.